aws/resource_aws_rds_cluster_instance_test.go

```diff
@@ -872,28 +872,70 @@ func TestAccAWSRDSClusterInstance_CACertificateIdentifier(t *testing.T) {
 	})
 }
 
+func testAccPreCheckAWSRDSClusterInstance_performanceInsights(t *testing.T) {
+	conn := testAccProvider.Meta().(*AWSClient).rdsconn
+
+	input := &rds.DescribeOrderableDBInstanceOptionsInput{
+		Engine:        aws.String("mysql"),
+		EngineVersion: aws.String("8.0.20"), // version cuts response time
+	}
+
+	supportsPerformanceInsights := false
+	err := conn.DescribeOrderableDBInstanceOptionsPages(input, func(resp *rds.DescribeOrderableDBInstanceOptionsOutput, lastPage bool) bool {
+		for _, instanceOption := range resp.OrderableDBInstanceOptions {
+			if instanceOption == nil {
+				continue
+			}
+
+			if aws.BoolValue(instanceOption.SupportsPerformanceInsights) {
+				supportsPerformanceInsights = true
+				return false // stop processing pages
+			}
+		}
+		return !lastPage
+	})
+
+	if err != nil {
+		t.Fatalf("unexpected PreCheck error: %s", err)
+	}
+
+	if !supportsPerformanceInsights {
+		t.Skipf("PreCheck fail: %s", "performance insights not supported, skipping acceptance test")
```

Comment:
Nit: Using the keyword "fail" when the test ultimately is "passing" (since skip == pass in Go testing) could be confusing

```suggestion
		t.Skipf("RDS Performance Insights not supported for engine (%s) version (%s), skipping acceptance test", engine, engineVersion)
```

Reply:
Fixed